Blog article: Preview of ytt Validations

pivotaljohn opened the pull request.

jtigger committed:
Draft

pivotaljohn commented:
Needs a few links to be updated, but copy ready for review. 👍🏻

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
````diff
@@ -0,0 +1,124 @@
+```
+
+That one new annotation — `@schema/validation` — will:
+- validate the final data value against that rule (`provider:` set to one of the three values),
````
```suggestion
- validate the final data value against that rule (`provider:` is set to one of the three values),
```

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
```diff
@@ -0,0 +1,124 @@
+
+### What are `ytt` Validations?
+
+Validations are constraints that you can declare on your Data Values to ensure that what's supplied is not just well-formed, but in the expected range of values.
```
```suggestion
Validations are constraints that you can define on Data Values via an annotation. A Validation ensures that a Data Value is in the expected range of values.
```

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
```diff
@@ -0,0 +1,124 @@
+
+For all the juicy details, check out the [Validations Proposal/Specification](https://github.com/vmware-tanzu/carvel/blob/004-schema-validation/proposals/ytt/004-schema-validation/README.md).
+
+This kind of thing is very useful for situations where a mis-configured deployment could waste a lot of resources and be much more difficult to troubleshoot than getting an immediate error message that the configuration value was wrong.
```
```suggestion
`ytt` Validations are very useful for situations where a mis-configured deployment could waste a lot of resources and be much more difficult to troubleshoot than getting an immediate error message that the configuration value was wrong.
```

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
```diff
@@ -0,0 +1,124 @@
+
+### How do I get started?
+
+`ytt` v0.41.0 will be released soon. When it does, grab yourself the latest.
```
Update with link once released.

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
```diff
@@ -0,0 +1,124 @@
+
+`ytt` v0.41.0 will be released soon. When it does, grab yourself the latest.
+
+Then, to enable Validations, and start including validation rules in a copy of _your_ schema:
```
```suggestion
Then, to enable Validations, start including validation rules in a copy of _your_ [Data Values Schema](how-to-write-schema.md):
```

pivotaljohn replied:
🙌🏻

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
````diff
@@ -0,0 +1,124 @@
+... and set the OS environment variable, `YTTEXPERIMENTS`...
+
+```console
+$ YTTEXPERIMENTS=validations yttt -f schema.yml -v dex.namespace=default
````
extra t in `ytt`. Happens in a couple other places as well.

pivotaljohn replied:
oops! Yeah, I have an alias `yttt` that executes my locally built version. 😬 

Will update.

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
```diff
@@ -0,0 +1,124 @@
+
+### Why are Validations Useful?
+
+To date, ytt Library / Carvel Package authors have had to "roll their own" logic to check that customer's inputs are valid: that data values are present/non-empty and within acceptable values.
```
customer or consumer?

pivotaljohn replied:
yeah... I thought "customer" would be a little more appropriate because it's about the relationship between the Package Author and the people using their stuff.... 

Thoughts?

gcheadle-vmware commented on site/content/blog/ytt-validations-preview.md:
```diff
@@ -0,0 +1,124 @@
+
+### How can I give feedback?
+
+We would love to tell you, thank you for asking!
```
```suggestion
We would love to tell you, thank you for asking! 😁 
```

pivotaljohn replied:
peerrrrfect!

gcheadle-vmware reviewed (commented):
Awesome write up! There are a few small changes I would make, otherwise my suggestions are to provide another voice for you. 

Also, I would leave a detail that the Playground with not have this since experiments are not enabled.

pivotaljohn committed:
Update site/content/blog/ytt-validations-preview.md

Co-authored-by: Garrett Cheadle <67297250+gcheadle-vmware@users.noreply.github.com>

pivotaljohn committed:
Apply suggestions from code review

Co-authored-by: Garrett Cheadle <67297250+gcheadle-vmware@users.noreply.github.com>

jtigger committed:
Include prompts for feedbacks

also:
- fix some typos in the examples
- tone-down the emoji game a little: fun, not silly.

jtigger committed:
Land a little more gracefully.

jtigger committed:
Include instructions for playground use

also:
- using the `version` command to verify an experiment is turned on.

jtigger committed:
Add reference to docs and backlog

microwavables commented on site/content/blog/ytt-validations-preview.md:
```diff
@@ -0,0 +1,153 @@
+
+The more time you spend doing this, the less time you spend making progress on your _actual_ goal.
+
+Of course the critical ingredient in all this is that the tool _actually_ solved the problem in the way you want it to. And this is it: here's your chance to add your perspective to the mix; a perspective we highly value. 🏆
```
```suggestion
Of course the critical ingredient in all this is that the tool _actually_ solved the problem in the way you want it to. And this is it: here's your chance to add your perspective to the mix; a perspective we highly value. 🏆

## Join the Carvel Community

We are excited to hear from you and learn with you! Here are several ways you can get involved:

* Join Carvel's slack channel, [#carvel in Kubernetes]({{% named_link_url "slack_url" %}}) workspace, and connect with over 1000+ Carvel users.
* Find us on [GitHub](https://github.com/vmware-tanzu/carvel). Suggest how we can improve the project, the docs, or share any other feedback.
* Attend our Community Meetings, happening every Thursday at 10:30 am PT / 1:30 pm ET. Check out the [Community page](/community/) for full details on how to attend.
```

gcheadle-vmware committed:
update validations docs link to develop branch
